miio/viomivacuum.py

```diff
@@ -227,96 +223,265 @@ def error(self) -> Optional[str]:
     @property
     def battery(self) -> int:
         """Battery in percentage."""
-        return self.data["battary_life"]
+        return self._data["battary_life"]
 
     @property
     def bin_type(self) -> ViomiBinType:
         """Type of the inserted bin."""
-        return ViomiBinType(self.data["box_type"])
+        return ViomiBinType(self._data["box_type"])
 
     @property
     def clean_time(self) -> timedelta:
         """Cleaning time."""
-        return pretty_seconds(self.data["s_time"])
+        return pretty_seconds(self._data["s_time"])
 
     @property
     def clean_area(self) -> float:
         """Cleaned area in square meters."""
-        return self.data["s_area"]
+        return self._data["s_area"]
 
     @property
     def fanspeed(self) -> ViomiVacuumSpeed:
         """Current fan speed."""
-        return ViomiVacuumSpeed(self.data["suction_grade"])
+        return ViomiVacuumSpeed(self._data["suction_grade"])
 
     @property
     def water_grade(self) -> ViomiWaterGrade:
         """Water grade."""
-        return ViomiWaterGrade(self.data["water_grade"])
+        return ViomiWaterGrade(self._data["water_grade"])
 
     @property
     def remember_map(self) -> bool:
         """True to remember the map."""
-        return bool(self.data["remember_map"])
+        return bool(self._data["remember_map"])
 
     @property
     def has_map(self) -> bool:
         """True if device has map?"""
-        return bool(self.data["has_map"])
+        return bool(self._data["has_map"])
 
     @property
     def has_new_map(self) -> bool:
         """TODO: unknown"""
-        return bool(self.data["has_newmap"])
+        return bool(self._data["has_newmap"])
 
     @property
     def mop_mode(self) -> ViomiMode:
         """Whether mopping is enabled and if so which mode
 
         TODO: is this really the same as mode?
         """
-        return ViomiMode(self.data["is_mop"])
+        return ViomiMode(self._data["is_mop"])
+
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self._data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self._data["hw_info"]
+
+    @property
+    def sw_info(self) -> str:
+        """SoftWare info."""
+        return self._data["sw_info"]
+
+    @property
+    def hypa_hours_left(self) -> int:
```

Comment:
Return the values for consumable `_left`s as `timedelta`s (to be consistent with other vacuums).

Reply:
fixed